examples/01_basic_ping_bot/src/main.rs

```diff
@@ -59,54 +58,74 @@ impl From<TungsteniteError> for Error {
 }
 
 fn main() {
-    tokio::run(try_main().map_err(|why| {
-        println!("Error running shard: {:?}", why);
-    }).boxed().compat(TokioDefaultSpawner));
+    tokio::run(main_future().map_err(|reason| {
+        eprintln!("Error when running shard {:?}:", reason);
+    }).boxed().compat());
 }
 
-async fn try_main() -> Result<(), Error> {
+async fn main_future() -> Result<(), Error> {
     env_logger::init();
-
     // Configure the client with your Discord bot token in the environment.
-    let token = env::var("DISCORD_TOKEN").expect("Expected a token in the environment");
-
-    // Create a new shard, specifying the token, the ID of the shard (0 of 1),
-    // and a handle to the event loop
-    let mut shard = await!(Shard::new(token, [0, 1]).compat())?;
-    println!("Connected shard");
-    let mut messages = shard.messages().compat();
+    let token = env::var("DISCORD_TOKEN").expect("No token was provided for the bot.");
+    // Create new shard.
+    let mut shard = await!(Shard::new(token, [0, 1]).compat()).unwrap();
+    let mut messages = shard.messages().expect("No shard messages found.").compat();
+    println!("Shard is connected.");
```

Comment:
This string doesn't need to change, and it makes more sense being immediately after the Shard instantiation than it does after instantiating the Shard _and_ taking the messages stream.